src/main/java/org/testng/internal/DynamicGraph.java

```diff
@@ -40,11 +40,27 @@ private Edge(int weight, T from, T to) {
     }
   }
 
+  public DynamicGraph() {
+    this(false);
+  }
+
+  DynamicGraph(boolean useLists) {
+    if (useLists) {
+      m_nodesFinished = Lists.newArrayList();
+      m_nodesReady = Lists.newArrayList();
+      m_nodesRunning = Lists.newArrayList();
+    } else {
+      m_nodesFinished = Sets.newLinkedHashSet();
```

Comment:
What if we always use HashSet?

Reply:
Where do we use `new DynamicGraph(true)`?

Reply:
> What if we always use HashSet?

The implementation is using HashSet only by default.

> Where do we use new DynamicGraph(true)?

We use `new DynamicGraph(true)` only in the test wherein we want to ensure that the time taken for the set implementation is lesser than the time taken for the list backed implementation.

Reply:
oops, bad eyes, sorry.

please, remove lists because you already proved that the change is better.

Reply:
Sure I will remove off the lists implementation as well.

Reply:
@juherr - btw, i just now realised.. If I remove off the lists support, then I will have to get rid of the `performanceTest()` test case as well, because now it wont have anything to assert against. Is that fine ?

Reply:
Sure. But if you really want to compare perfs in the test, you'll have to pass collections in the constructor (default constructor with Sets).